mettle/worker/worker.go

```diff
@@ -199,19 +210,20 @@ func runForever(instanceName, requestQueue, responseQueue, name, storage, dir, c
 		// possible for the subprocesses.
 		runtime.GC()
 		w.Report(true, false, true, "Awaiting next task...")
-		if _, err := w.RunTask(ctx); err != nil {
-			if ctx.Err() != nil {
-				// Error came from a signal triggered above. Give a brief period to send reports then die.
-				time.Sleep(500 * time.Millisecond)
-				return fmt.Errorf("terminated by signal")
-			}
+		_, err := w.RunTask(context.Background())
+		if err != nil {
```

Comment:
nit: `if _, err := w.RunTask(context.Background()); err != nil {`
It's better to scope errors within the statement if possible.

Also assume it's deliberate to change this not to propagate the context?

Reply:
Yes it is, as these changes mean it no longer cancels context to shut the worker down.

Reply:
I've now added the ctx back in (it's simpler than my `shutdownCh` idea after all!) so this takes in the context, as before.